src/index.ts

```diff
@@ -25,6 +25,22 @@ export interface Collection<K, V> extends Map<K, V> {
 export class Collection<K, V> extends Map<K, V> {
 	public static readonly default: typeof Collection = Collection;
 
+	/**
+	 * Obtains an element if the key exists, otherwise sets and returns the value provided by the default value generator.
+	 *
+	 * @param key The key to get from if it exists or set otherwise
+	 * @param defaultValueGenerator A function that returns the ensured value
+	 *
+	 * @example
+	 * collection.ensure(guildId, () => defaultGuildConfig);
+	 */
+	public ensure(key: K, defaultValueGenerator: (key: K, collection: this) => V): V {
+		if (this.has(key)) return this.get(key);
```

Comment:
```suggestion
		if (this.has(key)) return this.get(key)!;
```

Reply:
Thank you, I was just about to fix that!